flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/functions/PrintSinkFunctionTest.java

```diff
@@ -0,0 +1,330 @@
+					return null;
+				}
+
+				@Override
+				public Set<Entry<String, Future<Path>>> entrySet() {
+					return null;
+				}
+			};
+		}
+
+		@Override
+		public BroadcastVariableManager getBroadcastVariableManager() {
+			return null;
+		}
+
+		@Override
+		public AccumulatorRegistry getAccumulatorRegistry() {
+			return null;
+		}
+
+		@Override
+		public void acknowledgeCheckpoint(long checkpointId) {
+
+		}
+
+		@Override
+		public void acknowledgeCheckpoint(long checkpointId, StateHandle<?> state) {
+
+		}
+
+		@Override
+		public ResultPartitionWriter getWriter(int index) {
+			return null;
+		}
+
+		@Override
+		public ResultPartitionWriter[] getAllWriters() {
+			return new ResultPartitionWriter[0];
+		}
+
+		@Override
+		public InputGate getInputGate(int index) {
+			return null;
+		}
+
+		@Override
+		public InputGate[] getAllInputGates() {
+			return new InputGate[0];
+		}
+	};
+
+	public OutputStream out = new OutputStream() {
+		@Override
+		public void write(int b) throws IOException {
+
+		}
+	};
+
+	@Test
+	public void testPrintSinkStdOut(){
+
+		printStreamMock stream = new printStreamMock(out);
+		System.setOut(stream);
+
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		final HashMap<String, Accumulator<?, ?>> accumulators = new HashMap<String, Accumulator<?, ?>>();
+		final StreamingRuntimeContext ctx = new StreamingRuntimeContext(
+				this.envForPrefixNull,
+				executionConfig,
+				null,
+				null,
+				accumulators
+		);
+
+		PrintSinkFunction<String> printSink = new PrintSinkFunction<String>();
+		printSink.setRuntimeContext(ctx);
+		try {
+			printSink.open(new Configuration());
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		printSink.setTargetToStandardOut();
+		printSink.invoke("hello world!");
+
+		assertEquals("Print to System.out", printSink.toString());
+		assertEquals("hello world!", stream.result);
+
+		printSink.close();
+	}
+
+	@Test
+	public void testPrintSinkStdErr(){
+
+		printStreamMock stream = new printStreamMock(out);
+		System.setOut(stream);
+
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		final HashMap<String, Accumulator<?, ?>> accumulators = new HashMap<String, Accumulator<?, ?>>();
+		final StreamingRuntimeContext ctx = new StreamingRuntimeContext(
+				this.envForPrefixNull,
+				executionConfig,
+				null,
+				null,
+				accumulators
+		);
+
+		PrintSinkFunction<String> printSink = new PrintSinkFunction<String>();
+		printSink.setRuntimeContext(ctx);
+		try {
+			printSink.open(new Configuration());
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		printSink.setTargetToStandardErr();
+		printSink.invoke("hello world!");
+
+		assertEquals("Print to System.err", printSink.toString());
+		assertEquals("hello world!", stream.result);
+
+		printSink.close();
+	}
+
+	@Override
+	public void invoke(IN record) {
+
+	}
+
+	@After
+	public void restoreSystemOut() {
+		System.setOut(new PrintStream(new FileOutputStream(FileDescriptor.out)));
```

Comment:
Could you restore the original PrintStream here? You can save it in a variable like `PrintStream original = System.out` and restore it here.